_posts/2019-12-09-ebcdic-character-format-guide.markdown

```diff
@@ -31,52 +31,35 @@ This post is mainly going to deal with fixed width EBCDIC records.
 
 At it's heart EBCDIC is just a character encoding, in simple terms that is how we map a number onto character as computers only store numbers i.e. 61:a, 62:b, 63:c.
 
-There are multiple different layouts that are inoperable with each other, as they are all slightly different.
+There are multiple different versions of EBCDIC that are inoperable with each other, the different formats are defined by their code pages, and they are all different so you should check which one your project uses.
 
-EBCDIC contains data that has a schema, sort of data in a database record. Each record contains the data for a row in the table.
+Here is an example of a fixed width record (in ASCII for readability).
 
-The schemas are stored in separate files called copybooks, there are two main ways these records can be composed, either fixed or variable width records. The difference between these is how a record is formatted. A fixed width record will always have the exact same length e.g. fixed width of 512 bytes, means each record would be that length even if it was just a name. 
+{% highlight text %}
+00000001Niall     2
+00000002Martin    9
+{% endhighlight %}
 
-PICTURE HERE OF A FIXED WIDTH.?
+As you can see in the data, there is no schema baked into this record, so if you received this file you would not know how to parse it. 
 
-NUMBERS ARE STORED AS CHARs?
+The schemas are stored in separate files called copybooks, there are two main ways these records can be composed, either fixed or variable width records. The difference between these is how a record is formatted. A fixed width record will always have the exact same length e.g. fixed width of 512 bytes, means each record would be that length even if it was just a name. 
+
+Another thing to highlight, if you convert to ASCII the numbers just look like strings, rather than in a specific number format.
 
 #### EBCDIC Packing
 
-This is a method that EBCDIC uses to store the sign of a given number (if it's positive or negative).
+EBCDIC may store the numbers as strings, however sometimes it uses different characters than 0-9. These values are called packed and is how EBCDIC encodes a positive or negative number, if the schema defines the number to be packed. 
```

Comment:
This gives a confusing view of EBCDIC numbers and packing.
EBCDIC uses an 8 byte character set, meaning that every row of an EBCDIC file can be read as a string. With a strong focus on minimizing field widths, one trick that this format employs when encoding numbers is to have the sign (whether it's positive or negative) encoded as part of the string changing the least significant digit from a number to a non numeric character. This saves a character by turning the four character "-100" to the three character "10}"